### PR TITLE
Support coverage reports for projects not at root

### PR DESCRIPTION
**Pull request by adborbas**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires PATCH [version update](https://semver.org/)

### Context

Fixes the issue of coverage reporting failing for projects not in root.

Resolves: [STEP-592](https://bitrise.atlassian.net/browse/STEP-592)
Resolves: #18 

### Changes

- Propagate `project_location` input to coverage reporter.
- Restructured `bitrise.yml` to handle multiple e2e tests better.
- Remove unused `model_wrapper.go`.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -3,121 +3,142 @@ default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
...
+          description: |-
+            To prevent step testing issues, like referencing relative
+            files with just './some-file' in the step's code, which would
+            work for testing the step from this directory directly
+            but would break if the step is included in another `bitrise.yml`.
+          run_if: true
+          inputs:
+            - path: ${ORIGIN_SOURCE_DIR}/_tmp
+            - is_create_path: true
+      - script:
```
Comment (godrei) [inline bitrise.yml:57]: pls use the simple-git-clone step: https://github.com/bitrise-steplib/steps-deploy-to-itunesconnect-application-loader/blob/master/bitrise.yml#L114

File: bitrise.yml [hunk 2]
```diff
@@ -3,121 +3,142 @@ default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
...
+          title: Clone sample app
+          inputs:
+            - content: git clone $SAMPLE_APP_URL -b $SAMPLE_APP_BRANCH .
+      - flutter-installer:
+      - script:
```
Comment (godrei) [inline bitrise.yml:62]: why do we need this step?

File: coverage_executor.go [hunk 3]
```diff
@@ -2,30 +2,34 @@ package main
...
 }
 
 type realCoverageExecutor struct {
 	interrupt      interrupt
 	commandBuilder commandBuilder
 }
 
-func (r realCoverageExecutor) executeCoverage(additionalParams []string) bool {
-	coverageCmdModel := r.commandBuilder.buildCoverageCmd(additionalParams)
+func (r realCoverageExecutor) executeCoverage(cfg config, additionalParams []string) bool {
```
Comment (godrei) [inline coverage_executor.go:29]: why do we pass the whole config? why not just the ProjectLocation?

File: bitrise.yml (hunk 2 above)
Comment (adborbas) [reply bitrise.yml:62]: We do not need this, it was just a debugging element for me. I'll remove it.

File: coverage_executor.go (hunk 3 above)
Comment (adborbas) [reply coverage_executor.go:29]: Yeah you are right, we should just pass the location.

**Review by godrei — APPROVED** [review]

**Pull request merged**